app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -48,4 +47,14 @@ def host_config_params
 
     Host.new(organization: organization, location: location, hostgroup: host_group, operatingsystem: operatingsystem).params
   end
+
+  def find_smart_proxy
+    @smart_proxy = SmartProxy.authorized(:view_smart_proxies).find(registration_params['smart_proxy_id'])
+    features = @smart_proxy.features.map(&:name)
+
+    unless features.include?('Registration') && features.include?('Templates')
+      message = N_("Proxy doesn't have enabled one of the following features: 'Registration', 'Templates'")
```

Comment:
I'm not sure if lacks or misses is better, but this is shorter:
```suggestion
      message = N_("Proxy lacks one of the following features: 'Registration', 'Templates'")
```